src/requests/rocketchat.js

```diff
@@ -0,0 +1,194 @@
+		}
+	} catch (error) {
+		return handleError(error)
+	}
+}
+
+// Login function
+exports.login = async (username, password) => {
+	try {
+		const payload = { user: username, password }
+		const response = await chatPlatformAxios.post(apiEndpoints.ROCKETCHAT.LOGIN, payload)
+
+		return {
+			user_id: response.data.data.userId,
+			auth_token: response.data.data.authToken,
+		}
+	} catch (error) {
+		return handleError(error)
+	}
+}
+
+// Admin login function
+exports.adminLogin = async () => {
+	try {
+		const payload = {
+			user: process.env.CHAT_PLATFORM_ADMIN_EMAIL,
+			password: process.env.CHAT_PLATFORM_ADMIN_PASSWORD,
+		}
+		const response = await chatPlatformAxios.post(apiEndpoints.ROCKETCHAT.LOGIN, payload)
+		return response.data
+	} catch (error) {
+		return handleError(error)
+	}
+}
+
+// Initiate chat room function
+exports.initiateChatRoom = async (usernames, excludeSelf = true) => {
+	try {
+		const payload = { usernames: usernames.join(','), excludeSelf }
+		const response = await chatPlatformAxios.post(apiEndpoints.ROCKETCHAT.IM_CREATE, payload)
+		return {
+			room: {
+				room_id: response.data.room.rid,
+			},
+		}
+	} catch (error) {
+		throw handleError(error)
+	}
+}
+
+// Logout function
+exports.logout = async (userId, token) => {
+	try {
+		const response = await chatPlatformAxios.post(
+			apiEndpoints.ROCKETCHAT.LOGOUT,
+			{},
+			{
+				headers: {
+					'X-Auth-Token': token,
+					'X-User-Id': userId,
+				},
+			}
+		)
+		return response.data
+	} catch (error) {
+		return handleError(error)
+	}
+}
+
+// Logout other clients function
+exports.logoutOtherClients = async (userId, token) => {
+	try {
+		const response = await chatPlatformAxios.post(
+			apiEndpoints.ROCKETCHAT.LOGOUT_OTHER_CLIENTS,
+			{},
+			{
+				headers: {
+					'X-Auth-Token': token,
+					'X-User-Id': userId,
+				},
+			}
+		)
+		return response.data
+	} catch (error) {
+		return handleError(error)
+	}
+}
+
+// Send message function
+exports.sendMessage = async (username, password, rid, msg) => {
+	try {
+		const loginResponse = await this.login(username, password)
+
+		if (loginResponse.auth_token) {
+			const payload = {
+				message: {
+					rid,
+					msg,
+				},
+			}
+
+			const response = await chatPlatformAxios.post(apiEndpoints.ROCKETCHAT.CHAT_SEND_MESSAGE, payload, {
+				headers: {
+					'X-Auth-Token': loginResponse.auth_token,
+					'X-User-Id': loginResponse.user_id,
+				},
+			})
+
+			await this.logout(loginResponse.user_id, loginResponse.auth_token)
+
+			return response.data
+		} else {
+			throw new Error('Login failed, unable to send message')
+		}
+	} catch (error) {
+		return handleError(error)
+	}
+}
+
+// Set avatar function
+exports.setAvatar = async (username, imageUrl) => {
+	try {
+		const imageResponse = await axios.get(imageUrl, { responseType: 'arraybuffer' })
+		const imageBuffer = Buffer.from(imageResponse.data)
+		const imageBlob = new Blob([imageBuffer], { type: 'image/jpeg' })
```

Comment:
@nevil-mathew for png other type of image will this works ?

Reply:
It should, will test once we add images.